src/main/java/org/gridsuite/study/server/service/ConsumerService.java

```diff
@@ -398,367 +239,199 @@ public Consumer<Message<String>> consumeCaseImportFailed() {
         };
     }
 
-    @Bean
-    public Consumer<Message<String>> consumeSensitivityAnalysisResult() {
-        return message -> {
-            UUID resultUuid = UUID.fromString(message.getHeaders().get(RESULT_UUID, String.class));
-            String receiver = message.getHeaders().get(HEADER_RECEIVER, String.class);
-            if (receiver != null) {
-                NodeReceiver receiverObj;
-                try {
-                    receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
-
-                    LOGGER.info("Sensitivity analysis result '{}' available for node '{}'", resultUuid, receiverObj.getNodeUuid());
-
-                    // update DB
-                    updateSensitivityAnalysisResultUuid(receiverObj.getNodeUuid(), resultUuid);
+    /**
+     * parse the error message from the computation microservice and uses its data to notify the front
+     */
+    public void consumeCalculationFailed(Message<String> msg, ComputationType computationType) {
+        String receiver = msg.getHeaders().get(HEADER_RECEIVER, String.class);
+        String errorMessage = msg.getHeaders().get(HEADER_MESSAGE, String.class);
+        String userId = msg.getHeaders().get(HEADER_USER_ID, String.class);
+        UUID resultUuid = null;
+        // resultUuid is only used for the voltage initialization computation, I don't know why
+        if (computationType == VOLTAGE_INITIALIZATION) {
+            String resultId = msg.getHeaders().get(RESULT_UUID, String.class);
+            if (resultId != null) {
+                resultUuid = UUID.fromString(resultId);
+            }
+        }
+        if (!Strings.isBlank(receiver)) {
+            NodeReceiver receiverObj;
+            try {
+                receiverObj = objectMapper.readValue(URLDecoder.decode(receiver, StandardCharsets.UTF_8), NodeReceiver.class);
+
+                ComputationType updatedComputationType = computationType;
+                if (updatedComputationType == SHORT_CIRCUIT) {
+                    String busId = msg.getHeaders().get(HEADER_BUS_ID, String.class);
+                    if (!StringUtils.isEmpty(busId)) {
+                        updatedComputationType = SHORT_CIRCUIT_ONE_BUS;
+                    }
+                }
```

Comment:
maybe we could move this code block into `consumeShortCircuitAnalysisFailed` then we don't pollute generic code with some really shortcircuit specific code ?

Reply:
Very good idea. Done.